clustoapi/apps/attribute.py

```diff
@@ -177,14 +177,33 @@ def add_attr(name):
     if not obj:
         return util.dumps(msg, status)
 
-    for k in ('key', 'value'):
-        if k not in kwargs.keys():
-            bottle.abort(412, 'Provide at least "key" and "value"')
-
-    if 'number' in kwargs:
-        kwargs['number'] = int(kwargs['number'])
-
-    obj.add_attr(**kwargs)
+    try:
+        json_kwargs = request.json
+    except ValueError as ve:
+        return util.dumps('%s' % (ve,), 400)
+
+    if json_kwargs:
+        if request.query:
```

Comment:
The json body lives in request.forms. Changing to that wouldn't let you post any sort of body.